components/edit-collective/sections/UserSecurity.js

```diff
@@ -194,6 +212,143 @@ class UserTwoFactorAuth extends React.Component {
     }
   }
 
+  async setPassword() {
+    const { password, passwordKey, currentPassword, passwordScore } = this.state;
+
+    if (passwordScore <= 1) {
+      this.setState({
+        passwordError: (
+          <FormattedMessage defaultMessage="Password is too weak. Try to use more characters or use a password manager to generate a strong one." />
+        ),
+      });
+      return;
+    }
+
+    try {
+      this.setState({ passwordLoading: true });
+      await this.props.setPassword({
+        variables: { password, currentPassword },
+      });
+      this.props.addToast({
+        type: TOAST_TYPE.SUCCESS,
+        message: this.props.LoggedInUser.hasPassword ? (
+          <FormattedMessage defaultMessage="Password succesfully updated" />
+        ) : (
+          <FormattedMessage defaultMessage="Password succesfully set" />
+        ),
+      });
+      await this.props.refetchLoggedInUser();
+      this.setState({
+        currentPassword: '',
+        password: '',
+        passwordError: null,
+        passwordScore: null,
+        passwordLoading: false,
+        passwordKey: Number(passwordKey) + 1,
+      });
+    } catch (e) {
+      this.setState({ passwordError: e.message, passwordLoading: false });
+    }
+  }
+
+  renderPasswordManagement() {
+    const { LoggedInUser } = this.props;
+    const { password, passwordError, passwordLoading, passwordKey, currentPassword } = this.state;
+
+    return (
+      <Fragment>
+        <H3 fontSize="18px" fontWeight="700" mb={2}>
+          <FormattedMessage id="Password" defaultMessage="Password" />
+        </H3>
+        {passwordError && (
+          <MessageBox type="error" withIcon my={2} data-cy="password-error">
+            {passwordError}
+          </MessageBox>
+        )}
+        <Container mb="4">
+          <P py={2} mb={2}>
+            {LoggedInUser.hasPassword ? (
+              <FormattedMessage
+                id="Password.Change.Info"
+                defaultMessage="You already have a password set, you can change it using the following form."
+              />
+            ) : (
+              <FormattedMessage
+                id="Password.Set.Info"
+                defaultMessage="Setting a password is optional but this can be useful for regular users."
```

Comment:
I'm not convinced by this description, esp. the "regular users" part. Maybe:

```suggestion
                defaultMessage="Setting a password is optional but can be useful if you're using a password manager."
```

Or

```suggestion
                defaultMessage="Setting a password is optional but can be useful if you don't like magic links or if you prefer to use a password manager."
```